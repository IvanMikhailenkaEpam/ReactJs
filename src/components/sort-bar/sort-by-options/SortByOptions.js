import React from 'react';

const SortByOptions = ({onClick}) => {
  return (
    <div className="col-auto ml-auto">
      <span className="mr-3 font-weight-bold">Sort by</span>
      <span className="mr-5 font-weight-bold pointer" onClick={event => onClick(event, "release_date")}>
        release date
      </span>
      <span className="font-weight-bold pointer"
            onClick={event => onClick(event, "vote_average")}>rating</span>
    </div>
  );
};

export default SortByOptions;
