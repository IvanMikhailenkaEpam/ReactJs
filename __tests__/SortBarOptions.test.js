import React from 'react';
import { shallow } from 'enzyme';
import SortByOptions from "../src/components/sort-bar/sort-by-options/SortByOptions";

describe('SortByOptions', () => {
  it('renders SortByOptions correctly', () => {
    const component = shallow(<SortByOptions />);

    expect(component).toMatchSnapshot();
  });

  it('passed function for first span have been called', () => {
    const mockCallBack = jest.fn();
    const component = shallow(<SortByOptions  onClick={mockCallBack}/>);
    component.find("span.pointer").first().simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });

  it('passed function for second have been called', () => {
    const mockCallBack = jest.fn();
    const component = shallow(<SortByOptions  onClick={mockCallBack}/>);
    component.find("span.pointer").last().simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });
});
