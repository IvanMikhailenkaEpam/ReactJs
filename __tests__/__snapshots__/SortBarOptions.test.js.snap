// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortByOptions renders SortByOptions correctly 1`] = `
<div
  className="col-auto ml-auto"
>
  <span
    className="mr-3 font-weight-bold"
  >
    Sort by
  </span>
  <span
    className="mr-5 font-weight-bold pointer"
    onClick={[Function]}
  >
    release date
  </span>
  <span
    className="font-weight-bold pointer"
    onClick={[Function]}
  >
    rating
  </span>
</div>
`;

exports[`SortByOptions renders SortByOptions correctly 2`] = `
<div
  className="col-auto ml-auto"
>
  <span
    className="mr-3 font-weight-bold"
  >
    Sort by
  </span>
  <span
    className="mr-5 font-weight-bold pointer"
    onClick={[Function]}
  >
    release date
  </span>
  <span
    className="font-weight-bold pointer"
    onClick={[Function]}
  >
    rating
  </span>
</div>
`;
